PortalSwift/Sources/PortalSwift/Storage/ICloudStorage.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  ICloudStorage.swift
+//
+//  Created by Portal Labs, Inc.
+//  Copyright © 2022 Portal Labs, Inc. All rights reserved.
+//
+
+import CommonCrypto
+import Foundation
+
+/// A storage class that uses iCloud's key-value store to store/retrieve private keys.
+public class ICloudStorage: Storage, PortalStorage {
+  public enum ICloudStorageError: Error {
+    case noAPIKeyProvided(String)
+    case noAccessToICloud(String)
+    case notSignedIntoICloud(String)
+    case unableToDeriveKey(String)
+    case unableToRetrieveClient(String)
+    case failedValidateOperations(String)
+    case unknownError
+  }
+
+  public enum ICloudStatus: String {
+    case available
+    case notSignedIn
+    case noAccess
+  }
+
+  /// The Portal API instance to retrieve the client's and custodian's IDs.
+  public var api: PortalApi?
+  /// The key used to store the private key in iCloud.
+  public var key: String = ""
+
+  public let encryption: PortalEncryption
+
+  private let isSimulator = TARGET_OS_SIMULATOR != 0
+  private let storage: PortalKeyValueStore
```

Comment:
Where is PortalKeyValueStore defined? Im assuming that is specific to ICloud key value store

Reply:
I see it in utils

Reply:
Yeah, this is a new abstraction to make testing easier. It's just a passthrough to the iCloud Key Value Store